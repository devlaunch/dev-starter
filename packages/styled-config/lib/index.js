import { screenSize } from './config/screen-size';
export { screenSize };