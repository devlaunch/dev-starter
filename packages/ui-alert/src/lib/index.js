import { Alert } from "lib/components/Alert";

export { Alert };
