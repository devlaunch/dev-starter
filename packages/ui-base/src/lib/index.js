import { A } from "./components/Link/A";

import { Abbr } from "./components/Text/Abbr";
import { Address } from "./components/Text/Address";
import { B } from "./components/Text/B";
import { Blockquote } from "./components/Text/Blockquote";
import { Code } from "./components/Text/Code";
import { Display1 } from "./components/Text/Display1";
import { Display2 } from "./components/Text/Display2";
import { Display3 } from "./components/Text/Display3";
import { Display4 } from "./components/Text/Display4";
import { H1 } from "./components/Text/H1";
import { H2 } from "./components/Text/H2";
import { H3 } from "./components/Text/H3";
import { H4 } from "./components/Text/H4";
import { H5 } from "./components/Text/H5";
import { H6 } from "./components/Text/H6";
import { Hgroup } from "./components/Text/Hgroup";
import { Initialism } from "./components/Text/Initialism";
import { Kbd } from "./components/Text/Kbd";
import { Lead } from "./components/Text/Lead";
import { Mark } from "./components/Text/Mark";
import { P } from "./components/Text/P";
import { Pre } from "./components/Text/Pre";
import { Samp } from "./components/Text/Samp";
import { Small } from "./components/Text/Small";
import { Strong } from "./components/Text/Strong";
import { Sub } from "./components/Text/Sub";
import { Sup } from "./components/Text/Sup";

import { Article } from "./components/Articles/Article";
import { Aside } from "./components/Articles/Aside";
import { Main } from "./components/Articles/Main";
import { Nav } from "./components/Articles/Nav";
import { Footer } from "./components/Articles/Footer";
import { Header } from "./components/Articles/Header";
import { Summary } from "./components/Articles/Summary";
import { Section } from "./components/Articles/Section";

import { Button } from "./components/Form/Button";
import { Fieldset } from "./components/Form/Fieldset";

import { Caption } from "./components/Tables/Caption";
import { Table } from "./components/Tables/Table";
import { Textarea } from "./components/Form/Textarea";
import { Th } from "./components/Tables/Th";

import { Dd } from "./components/Lists/Dd";
import { Dfn } from "./components/Lists/Dfn";
import { Dl } from "./components/Lists/Dl";
import { Dt } from "./components/Lists/Dt";
import { Ol } from "./components/Lists/Ol";
import { Ul } from "./components/Lists/Ul";

import { Figcaption } from "./components/Image/Figcaption";
import { Figure } from "./components/Image/Figure";
import { Img } from "./components/Image/Img";
import { Svg } from "./components/Image/Svg";

import { Input } from "./components/Form/Input";
import { Legend } from "./components/Form/Legend";
import { Label } from "./components/Form/Label";
import { Optgroup } from "./components/Form/Optgroup";
import { Output } from "./components/Form/Output";
import { Progress } from "./components/Form/Progress";
import { Select } from "./components/Form/Select";

import { Box } from "./components/Layout/Box";
import { Flex } from "./components/Layout/Flex";

import { BootstrapBaseCss } from "./components/BootstrapBaseCss";
import { Hr } from "./components/Hr";

export {
  A,
  Abbr,
  Address,
  Article,
  Aside,
  B,
  Blockquote,
  Button,
  Caption,
  Code,
  Dd,
  Dfn,
  Display1,
  Display2,
  Display3,
  Display4,
  Dl,
  Dt,
  Fieldset,
  Figcaption,
  Figure,
  Footer,
  BootstrapBaseCss,
  H1,
  H2,
  H3,
  H4,
  H5,
  H6,
  Header,
  Hgroup,
  Hr,
  Img,
  Input,
  Initialism,
  Kbd,
  Lead,
  Legend,
  Label,
  Mark,
  Main,
  Nav,
  Ol,
  Optgroup,
  Output,
  P,
  Pre,
  Progress,
  Samp,
  Section,
  Select,
  Small,
  Strong,
  Sub,
  Summary,
  Sup,
  Svg,
  Table,
  Textarea,
  Th,
  Ul,
  Box,
  Flex
};
