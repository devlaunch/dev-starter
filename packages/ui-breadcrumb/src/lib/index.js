import { Breadcrumb } from "./components/Breadcrumb";
import { BreadcrumbItem } from "./components/BreadcrumbItem";

export { Breadcrumb, BreadcrumbItem };
