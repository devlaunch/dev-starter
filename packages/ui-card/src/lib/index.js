import { Card } from './components/Card';
import { CardBody } from './components/CardBody';
import { CardFooter } from './components/CardFooter';
import { CardHeader } from './components/CardHeader';
import { CardImageHeader } from './components/CardImageHeader';
import { CardText } from './components/CardText';
import { CardTitle } from './components/CardTitle';

export {
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  CardImageHeader,
  CardText,
  CardTitle,
};
