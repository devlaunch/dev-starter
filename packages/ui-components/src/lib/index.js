import { Alert } from "@devlaunch/ui-alert";
import { Badge } from "@devlaunch/ui-badge";
import {
  A,
  Abbr,
  Address,
  Article,
  Aside,
  B,
  Blockquote,
  Button as BaseButton,
  Caption,
  Code,
  Dd,
  Dfn,
  Dialog,
  Dl,
  Dt,
  Fieldset,
  Figcaption,
  Figure,
  Footer,
  H1,
  H2,
  H3,
  H4,
  H5,
  H6,
  Header,
  Hgroup,
  Hr,
  Img,
  Input,
  Kbd,
  Legend,
  Label,
  Main,
  Nav as Navigation,
  Ol,
  Optgroup,
  Output,
  P,
  Pre,
  Progress,
  Samp,
  Section,
  Select,
  Small,
  Strong,
  Sub,
  Summary,
  Sup,
  Svg,
  Table as BaseTable,
  Textarea,
  Th,
  Ul
} from "@devlaunch/ui-base";
import { Breadcrumb, BreadcrumbItem } from "@devlaunch/ui-breadcrumb";
import { Button, ButtonGroup, ButtonToolbar } from "@devlaunch/ui-button";
import {
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  CardImageHeader,
  CardText,
  CardTitle
} from "@devlaunch/ui-card";
import {
  theme,
  border,
  borderRadius,
  boxShadow,
  colors,
  fontFamily,
  fontSize,
  fontWeight,
  height,
  margin,
  padding,
  screenSize,
  width
} from "@devlaunch/ui-config";
import { Container } from "@devlaunch/ui-container";
import {
  Dropdown,
  DropdownItem,
  DropdownMenu,
  DropdownDivider
} from "@devlaunch/ui-dropdown";
import {
  FormCheck,
  FormCheckInput,
  FormControl,
  FormControlFile,
  FormControlPlainText,
  FormGroup,
  FormInline,
  FormText,
  InputGroup,
  InputGroupAppend,
  InputGroupPrepend,
  InputGroupText
} from "@devlaunch/ui-form";
import { Column, Row } from "@devlaunch/ui-grid-system";
import { Jumbotron } from "@devlaunch/ui-jumbotron";
import { ListGroup, ListGroupItem } from "@devlaunch/ui-listgroup";
import {
  Modal,
  ModalBody,
  ModalContent,
  ModalDialog,
  ModalFooter,
  ModalHeader,
  ModalTitle
} from "@devlaunch/ui-modal";
import { Nav, NavLink } from "@devlaunch/ui-nav";
import { Navbar, NavbarLink } from "@devlaunch/ui-navbar";
import {
  Popover,
  PopoverArrow,
  PopoverBody,
  PopoverHeader
} from "@devlaunch/ui-popover";
import { Table, Tr } from "@devlaunch/ui-table";
import { Tooltip, TooltipArrow, TooltipInner } from "@devlaunch/ui-tooltip";

export {
  Alert,
  Badge,
  A,
  Abbr,
  Address,
  Article,
  Aside,
  B,
  Blockquote,
  BaseButton,
  Caption,
  Code,
  Dd,
  Dfn,
  Dialog,
  Dl,
  Dt,
  Fieldset,
  Figcaption,
  Figure,
  Footer,
  H1,
  H2,
  H3,
  H4,
  H5,
  H6,
  Header,
  Hgroup,
  Hr,
  Img,
  Input,
  Kbd,
  Legend,
  Label,
  Main,
  Navigation,
  Ol,
  Optgroup,
  Output,
  P,
  Pre,
  Progress,
  Samp,
  Section,
  Select,
  Small,
  Strong,
  Sub,
  Summary,
  Sup,
  Svg,
  BaseTable,
  Textarea,
  Th,
  Ul,
  Breadcrumb,
  BreadcrumbItem,
  Button,
  ButtonGroup,
  ButtonToolbar,
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  CardImageHeader,
  CardText,
  CardTitle,
  theme,
  border,
  borderRadius,
  boxShadow,
  colors,
  fontFamily,
  fontSize,
  fontWeight,
  height,
  margin,
  padding,
  screenSize,
  width,
  Container,
  Dropdown,
  DropdownItem,
  DropdownMenu,
  DropdownDivider,
  FormCheck,
  FormCheckInput,
  FormControl,
  FormControlFile,
  FormControlPlainText,
  FormGroup,
  FormInline,
  FormText,
  InputGroup,
  InputGroupAppend,
  InputGroupPrepend,
  InputGroupText,
  Column,
  Row,
  Jumbotron,
  ListGroup,
  ListGroupItem,
  Modal,
  ModalBody,
  ModalContent,
  ModalDialog,
  ModalFooter,
  ModalHeader,
  ModalTitle,
  Nav,
  NavLink,
  Navbar,
  NavbarLink,
  Popover,
  PopoverArrow,
  PopoverBody,
  PopoverHeader,
  Table,
  Tr,
  Tooltip,
  TooltipArrow,
  TooltipInner
};
