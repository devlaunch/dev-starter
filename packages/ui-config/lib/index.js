

Object.defineProperty(exports, '__esModule', {
  value: true,
});
exports.default = void 0;

const _screenSize = _interopRequireDefault(require('./config/screen-size'));

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj };
}

const _default = _screenSize.default;
exports.default = _default;
