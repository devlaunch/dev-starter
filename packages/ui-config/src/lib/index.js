import { theme } from "./theme";
import {
  border,
  borderRadius,
  boxShadow,
  colors,
  fontFamily,
  fontSize,
  fontWeight,
  height,
  margin,
  padding,
  screenSize,
  width,
  themeGet
} from "./utils";

export {
  theme,
  themeGet,
  // utils
  border,
  borderRadius,
  boxShadow,
  colors,
  fontFamily,
  fontSize,
  fontWeight,
  height,
  margin,
  padding,
  screenSize,
  width
};
