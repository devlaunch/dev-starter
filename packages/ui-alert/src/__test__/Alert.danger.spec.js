import React from "react";
import { mount } from "enzyme";

import { Alert } from "lib";

test("Styles matches danger", () => {
  const component = mount(<Alert modifiers={["danger"]} />);
  expect(component).toMatchSnapshot();
  expect(component).toHaveStyleRule("color", "#721c24");
  expect(component).toHaveStyleRule("background-color", "#f8d7da");
  expect(component).toHaveStyleRule("border-color", "#f5c6cb");
  expect(component).toHaveStyleRule("border-top-color", "#f1b0b7", {
    modifier: "& > hr"
  });
  expect(component).toHaveStyleRule("color", "#491217", {
    modifier: "& > a"
  });
  expect(component).toHaveStyleRule("color", "#491217", {
    modifier: "& > a:hover"
  });
});
