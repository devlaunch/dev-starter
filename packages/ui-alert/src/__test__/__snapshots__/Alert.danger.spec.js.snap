// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styles matches danger 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  margin-bottom: 1rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: #721c24;
  background-color: #f8d7da;
  border-color: #f5c6cb;
}

.c0 > a {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

.c0 > hr {
  border-top-color: #f1b0b7;
}

.c0 > a,
.c0 > a:hover {
  color: #491217;
}

<Alert
  modifiers={
    Array [
      "danger",
    ]
  }
>
  <styled.div
    modifiers={
      Array [
        "danger",
      ]
    }
  >
    <div
      className="c0"
    />
  </styled.div>
</Alert>
`;
